### DISCUSSION TITLE
SBT manual compilation

### DISCUSSION THREAD

**Discussion by tdroxler** [scalameta/metals#3696] [Q&A] [answered]
Hello everyone,

I'm trying metals with sbt and I would like to know if it's possible to disable the auto compilation from metals?
There are 2 cases, when I simply write `def`, I directly receive a diagnostic: ` = expected ...`, while I obviously know it's not correct yet. Can we disable this?

The other thing I want to disable is the `compile-on-save`, I prefer to manually enter `compile` in my sbt console when I know I want to compile and get the potential diagnostics.

Is it doable?

Thx, cheers

**Comment by tgodzik** [comment, marked as answer]
> I'm trying metals with sbt and I would like to know if it's possible to disable the auto compilation from metals?
There are 2 cases, when I simply write def, I directly receive a diagnostic: = expected ..., while I obviously know it's not correct yet. Can we disable this?

Syntax errors are shown right away and unfortunately there is no way to turn that off.

> The other thing I want to disable is the compile-on-save, I prefer to manually enter compile in my sbt console when I know I want to compile and get the potential diagnostics.

That's also not possible, Metals relies on compilation to refresh indexes so we try to compile as soon of possible because of that.

**Reply by tdroxler** [reply]
alright, thx for your answer
